### PR TITLE
Add Doctor Class

### PR DESCRIPTION
**Pull request by Mohammed-Faizzzz**
Added Doctor Class and Integrated it with the pre-existing classes/modifications. There are a few kinks to sort out:
* height adjustment on the GUI: currently, there is an issue with the GUI and there is a white box at the bottom. It shouldn't be much of an issue
* Also, I'll be including the method calls such that when a patient is added into the addressbook, the doctor's patients list automatically gets updated.
* Testcases and Codecov: I might need some help with this

### CODE REVIEW

**Review by kohkaijie — APPROVED** [review]
LGTM once we fix the fix the merge conflict regarding the JsonSerialisableAddressBook

**Review by cmHuang777 — APPROVED** [review]
Overall looks good, may be we will merge first then fix the bugs in tests later. Kai jie will be doing it!

**Pull request merged**
